pkg/controller/service/service_controller.go

```diff
@@ -559,7 +559,9 @@ func portSlicesEqualForLB(x, y []*v1.ServicePort) bool {
 }
 
 func portEqualForLB(x, y *v1.ServicePort) bool {
-	// TODO: Should we check name?  (In theory, an LB could expose it)
+	// Note: Fields for uniqueness checking include Name、Protocol and NodePort
+	// Name(In theory, an LB could expose it, so we check it)
+	// TargetPort(It is not relevant for load balancing, so we ignore it)
```

Comment:
This is a little confusing, I would put the comment below `if x.Port != y.Port { ...`

`// Ignore differences in TargetPort as it is not relevant for load balancing.